pkg/tsdb/loki/frame.go

```diff
@@ -0,0 +1,121 @@
+package loki
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// we adjust the dataframes to be the way frontend & alerting
+// wants them.
+func adjustFrame(frame *data.Frame, query *lokiQuery) *data.Frame {
+	labels := getFrameLabels(frame)
+
+	timeFields, nonTimeFields := partitionFields(frame)
+
+	isMetricFrame := nonTimeFields[0].Type() != data.FieldTypeString
+
+	isMetricRanged := isMetricFrame && query.QueryType == QueryTypeRange
+
+	name := formatName(labels, query)
+	frame.Name = name
+
+	if frame.Meta == nil {
+		frame.Meta = &data.FrameMeta{}
+	}
+
+	if isMetricRanged {
+		frame.Meta.ExecutedQueryString = "Expr: " + query.Expr + "\n" + "Step: " + query.Step.String()
+	} else {
+		frame.Meta.ExecutedQueryString = "Expr: " + query.Expr
+	}
+
+	if !isMetricFrame {
+		frame.Meta.PreferredVisualization = data.VisTypeLogs
+	}
+
+	for _, field := range timeFields {
+		field.Name = "time"
+
+		if isMetricRanged {
+			if field.Config == nil {
+				field.Config = &data.FieldConfig{}
+			}
+			field.Config.Interval = float64(query.Step.Milliseconds())
+		}
+	}
+
+	for _, field := range nonTimeFields {
+		field.Name = "value"
+		if field.Config == nil {
+			field.Config = &data.FieldConfig{}
+		}
+		field.Config.DisplayNameFromDS = name
+	}
+
+	return frame
+}
+
+func formatNamePrometheusStyle(labels map[string]string) string {
+	metricName := labels["__name__"]
```

Comment:
I think we can remove this part with metricName because loki metrics result don't have metrics in the way prometheus does.

Reply:
good catch! done.